library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -732,7 +814,10 @@ public void onLoadCompleted(Chunk loadable, long elapsedRealtimeMs, long loadDur
 
   @Override
   public void onLoadCanceled(
-      Chunk loadable, long elapsedRealtimeMs, long loadDurationMs, boolean released) {
+      Chunk loadable, long elapsedRealtimeMs, long loadDurationMs, boolean released, boolean interrupted) {
+    if (interrupted) {
+      discardAfterInterrupt();
```

Comment:
The onLoadCanceled method does 3 things:
 1. Report event to eventDispatcher.
 2. Update sample queues as needed to continue loading afterwards
 3. Trigger or request to continue loading.
And it used for 2 main cases now: 
 A. Cancelation after seek or disabling tracks.  -> clear sample queues completely.
 B. Cancelation after reevaluateBuffer.  -> discard only pending load upstream.

Could you make sure both cases (A and B) are handled in the same way in the same order? At least step 1 can share the same code. Step 2 is intentionally different and this is where I would expect the "if(interrupted) { doB() } else { doA(); } code. And step 3 can be shared again.